spec/unit/lib/temporal/activity/context_spec.rb

```diff
@@ -90,17 +90,18 @@
   end
 
   describe '#last_heartbeat_throttled' do
-    before { allow(client).to receive(:record_activity_task_heartbeat).and_return(heartbeat_response) }
+    before { allow(connection).to receive(:record_activity_task_heartbeat).and_return(heartbeat_response) }
 
-    let(:metadata_hash) { Fabricate(:activity_metadata, heartbeat_timeout: 10).to_h }
+    let(:metadata_hash) { Fabricate(:activity_metadata, heartbeat_timeout: 3).to_h }
 
     it 'true when throttled, false when not' do
       subject.heartbeat(iteration: 1)
       expect(subject.last_heartbeat_throttled).to be(false)
       subject.heartbeat(iteration: 2)
       expect(subject.last_heartbeat_throttled).to be(true)
-      subject.heartbeat(iteration: 3)
-      expect(subject.last_heartbeat_throttled).to be(true)
+
+      # Shutdown to drain remaining threads
+      heartbeat_thread_pool.shutdown
```

Comment:
This code was raising some sort of error that was previously being ignored. Now that unhandled errors result in crashes to the process, this needs to be properly shutdown so that threads encountering bad test state don't fail.